### PR TITLE
Rename sbt plugin artifact so it's consistent with other artifacts

### PR DESCRIPTION
**Pull request by mdedetrich**
Currently the artifact name for the sbt plugin is inconsistent with the other artifacts that pekko-grpc generates, i.e. currently the sbt artifact name is `sbt-pekko-grpc` where as all other artifacts start with `pekko-grpc-<REST>`. This even includes the other build tool plugins that pekko-grpc provides, i.e. `pekko-grpc-gradle-plugin`/`pekko-grpc-maven-plugin`

### CODE REVIEW

**Comment by pjfanning** [comment]
I don't think this is necessary. I'm still going to fight to get the RC1 approved.

**Comment by mdedetrich** [comment]
> I don't think this is necessary. I'm still going to fight to get the RC1 approved.

I would prefer to get this done because we can't change it later easily without breaking users

**Review by pjfanning — CHANGES_REQUESTED** [review]
-1 from me

We have a viable RC in progress and the name matches the akka one

**Comment by mdedetrich** [comment]
> -1 from me
> 
> We have a viable RC in progress and the name matches the akka one

Im sorry but its more important to get the naming for the artifacts correct and consistent over a possible saving of a single day on an RC with only a negative vote.

File: docs/src/main/paradox/release-notes/index.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ We haven't had to fix any significant bugs that were in Akka gRPC 2.1.6.
   `mavenCentral()` to the `pluginManagement`'s `repositories` entry. See
   [Installation docs](https://pekko.apache.org/docs/pekko-grpc/current/buildtools/gradle.html#installation) for more
   info.
+* The naming convention of the Pekko gRPC sbt plugin has changed, i.e. where
+  as with Akka the artifact was named `sbt-akka-grpc` with Pekko its named
+  `pekko-grpc-sbt-plugin` so its consistent with `pekko-grpc-gradle-plugin`/`pekko-grpc-maven-plugin`.
```
Comment (pjfanning) [inline docs/src/main/paradox/release-notes/index.md:35]: should be `it's` twice - both cases are short for `it is`

**Review by pjfanning — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/release-notes/index.md (hunk 1 above)
Comment (mdedetrich) [reply docs/src/main/paradox/release-notes/index.md]: Fixed.

**Comment by mdedetrich** [comment]
@pjfanning PR passing and the grammer fixes you have pointed out has been applied

**Comment by pjfanning** [comment]
I'm still pondering this. I think we might need a vote on the dev list.

**Comment by mdedetrich** [comment]
The arguments from my end should be clear, unlike all of our other artifacts which start with pekko (this can clearly be seen at https://repository.apache.org/content/groups/staging/org/apache/pekko/) currently the sbt-plugin is an exception. I also don't know if there is an ASF rule regarding artifact naming needing to start with `pekko`, but there is one with github repo's (i.e. all of our repos need to start with `incubating-pekko`)

In other words it sticks out with a giant sore thumb and since we are dealing with artifact naming it should be fixed before the first release to avoid unnecessary disruption later (this should be obvious). 

If you think its necessary then go ahead but I hardly see how this is that controversial.

**Review by gmethvin — APPROVED** [review]

**Comment by pjfanning** [comment]
Up until today, all the Pekko artifacts have been named after the Akka ones that they are based on.

This artifact is based on https://central.sonatype.com/artifact/com.lightbend.akka.grpc/sbt-akka-grpc

Why should we break the pattern for this?

**Comment by mdedetrich** [comment]
> Up until today, all the Pekko artifacts have been named after the Akka ones that they are based on.

No they haven't, the pekko-connectors for example is entirely different from the akka ones (which used to be called alpakka)

> Why should we break the pattern for this

I already provided the arguments before

**Comment by pjfanning** [comment]
> > Up until today, all the Pekko artifacts have been named after the Akka ones that they are based on.
> 
> No they haven't, the pekko-connectors artifact naming for example is entirely different from the akka ones (which used to be called alpakka)
> 
> > Why should we break the pattern for this
> 
> I already provided the arguments before

The pekko connectors name change was agreed many many months. It wasn't decided on just as we attempted to do releases.

**Comment by mdedetrich** [comment]
> Changing this name is going to cause major downstream disruption in builds that use pekko-grpc.

> These include builds include

> incubator-pekko-connectors
> incubator-pekko-samples

These are unreleased

> Still -1 from me

Then put it up on the mailing list, this PR already got an approval from someone else.

**Comment by mdedetrich** [comment]
Also as far as I know, there was no real community decision to rename Alpakka to Connectors, at least I don't remember voting on it. It was just done (since Alpakka didn't make sense in the context of Pekko, it being a pun on Akka) and no one happened to complain about the name.

**Comment by gmethvin** [comment]
I prefer the naming actually be consistent.

I guess one argument for keeping it the same as Akka is that it makes updating artifact names slightly easier when migrating a project to Pekko, but this is a very small one time cost.

**Comment by pjfanning** [comment]
Ok. I'll remove my -1 since we need to go ahead and do an RC2. @mdedetrich can you fix up the other repos that I mentioned when we get a new grpc snapshot?

**Review by pjfanning — COMMENTED** [review]
remove -1

**Comment by mdedetrich** [comment]
> Ok. I'll remove my -1 since we need to go ahead and do an RC2. @mdedetrich can you fix up the other repos that I mentioned when we get a new grpc snapshot?

Will do, I'll generate new snapshots just after this merge and update downstream dependencies.

**Comment by mdedetrich** [comment]
New version with artifact changes is 1.0.0-RC1-3-ae23c14d-SNAPSHOT

* pekko-connectors PR at https://github.com/apache/incubator-pekko-connectors/pull/217
* pekko-samples PR at https://github.com/apache/incubator-pekko-samples/pull/62

I have also updated the `Pekko 1.0.1 & Pekko HTTP 1.0.0 & Pekko Connectors Kafka 1.0.0` entry at https://cwiki.apache.org/confluence/display/PEKKO/Testing+with+Pekko+Snapshot+Jars
